File: UnionPGE/Source.cpp
#include <iostream>
#include <stdlib.h>
#include <string>
#include <vector>
#include <typeinfo>
#include <thread>
#include <mutex>
#include <chrono>
using namespace std::chrono_literals;

#define OLC_PGE_APPLICATION
#include "olcPixelGameEngine.h"
class App : public olc::PixelGameEngine
{
public:
	App(std::string name)
	{
		sAppName = name;
	}

public:
	bool OnUserCreate() override;

	bool OnUserUpdate(float fElapsedTime) override;
};


std::string __uni__arrow(std::string* target, std::string object, bool dir) {
	if (dir)
		* target = object + *target;
	else
		*target += object;
	return *target;
}
int __uni__arrow(int* target, int object, bool dir) { return *target += object; }
std::string __uni__arrow(std::string* target, int object, bool dir) {
	if (dir)
		* target = std::to_string(object) + *target;
	else
		*target += std::to_string(object);
	return *target;
}


float angle(0);
int width; int height;
bool App::OnUserCreate() {
	width = ScreenWidth();
	height = ScreenHeight();
	return true;
}
bool App::OnUserUpdate(float fElapsedTime) {
	float ca(cos((angle + 100)));
	float cb(sin(angle));
	(angle += 0.02);
	Clear(olc::BLACK);
	float w(5);
	//float w((float)GetMouseY()/(float)height*10.f);
	float h(((w * height) / width));
	float xmin((-1 * w / 2));
	float ymin((-1 * h / 2));
	float maxIterations(100);
	float xmax((xmin + w));
	float ymax((ymin + h));
	float dx(((xmax - xmin) / width));
	float dy(((ymax - ymin) / height));
	float x(xmin);
	for (int i = 0; i < width; i++) {
		float y(ymin);
		for (int j = 0; j < height; j++) {
			float a(x);
			float b(y);
			float n(0);
			while ((n < maxIterations)) {
				float aa((a * a));
				float bb((b * b));
				if (((aa + bb) > 4)) {
					break;
				}
				float twoab((2 * a * b));
				a = (aa - bb + ca);
				b = (twoab + cb);
				n += 1;
			}
			olc::Pixel pix;
			if ((n == maxIterations)) {
				pix = olc::Pixel(olc::BLACK);
			}
			else {
				float hu((n / maxIterations * 255) + 20);
				pix = olc::Pixel(hu*1.2, hu*.8, hu);
			}
			Draw(i, j, pix);
			(y += dy);
		}
		(x += dx);
	}
	return true;
}
int main() {
	App demo("Julia Set");
	if (demo.Construct(640, 360, 1, 1)) {
		demo.Start();
	}
	return 0;
}

File: source.go
using olcPixelGameEngine


angle: float(0)
width:int; height:int;

func: bool App::OnUserCreate:
	width = ScreenWidth()
	height = ScreenHeight()
	return true

func: bool App::OnUserUpdate(fElapsedTime: float):

	ca: float = cos(angle + 100)
	cb: float = sin(angle)

	angle += 0.03

	Clear(olc::BLACK)

	w: float = 5
	h: float = (w*height) / width

	xmin: float = -1*w/2
	ymin: float = -1*h/2

	maxIterations: float (100)

	xmax: float = xmin + w
	ymax: float = ymin + h

	dx: float = (xmax - xmin) / width
	dy: float = (ymax - ymin) / height

	x: float = xmin
	for i to width:
		y: float = ymin
		for j to height:
			a: float = x
			b: float = y
			n: float (0)
			while n < maxIterations:
				aa: float = a * a
				bb: float = b * b
				if aa + bb > 4:
					break
				twoab: float = 2 * a * b
				a = aa - bb + ca
				b = twoab + cb
				n <- 1
			pix: olc::Pixel;
			if n == maxIterations:
				pix = olc::Pixel(olc::BLACK)
			else:
				hu: float =  n / maxIterations * 255 + 50
				pix = olc::Pixel(hu*1.2, hu*0.8, hu)
			Draw(i, j, pix)
			y += dy
		x += dx

	return true



func: int main:

	demo: App ("Julia Set")
	if demo.Construct(640, 360, 1, 1):
		demo.Start()

	return 0